README.md
# Steps package for Flutter

[![pub package](https://img.shields.io/pub/v/steps.svg)](https://pub.dartlang.org/packages/steps)

Flutter plugin for displaying vertical/horizontal steps with other Flutter widgets on Android and iOS.

![vertical with rich content](https://github.com/jamalbelilet/steps/blob/master/doc/vertical.png?raw=true)

![horizontal with rich content](https://github.com/jamalbelilet/steps/blob/master/doc/horizontal.png?raw=true)

*Note*: This package is still under development.
[Feedback welcome](https://github.com/jamalbelilet/steps/issues) and
[Pull Requests](https://github.com/jamalbelilet/steps/pulls) are most welcome!

## Installation

First, add `steps` as a [dependency in your pubspec.yaml file](https://flutter.io/using-packages/).

### Example

```dart
import 'package:flutter/material.dart';
import 'package:steps/steps.dart';

void main() => runApp(MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Home(title: 'Steps'),
    ));

class Home extends StatelessWidget {
  final String title;
  Home({Key key, this.title}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text(title),
        ),
        body: Container(
          alignment: Alignment.topCenter,
          child: Steps(
            direction: Axis.vertical,
            size: 20.0,
            path: {'color': Colors.lightBlue.shade200, 'width': 3.0},
            steps: [
              {
                'color': Colors.white,
                'background': Colors.lightBlue.shade200,
                'label': '1',
                'content': Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: <Widget>[
                    Text(
                      'Laborum exercitation',
                      style: TextStyle(fontSize: 22.0),
                    ),
                    Text(
                      'Qui et consectetur esse duis excepteur magna consectetur.',
                      style: TextStyle(fontSize: 12.0),
                    ),
                  ],
                ),
              },
              {
                'color': Colors.white,
                'background': Colors.lightBlue.shade700,
                'label': '2',
                'content': Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: <Widget>[
                    Text(
                      'Laborum exercitation est veniam',
                      style: TextStyle(fontSize: 22.0),
                    ),
                    Image.asset(
                      'assets/cat.jpg',
                      width: 250,
                      height: 120,
                      fit: BoxFit.cover,
                      alignment: Alignment.center,
                    ),
                    Text(
                      '''
Occaecat qui do mollit
Adipisicing reprehenderit deserunt mollit
Quis officia adipisicing aute
                      ''',
                      style: TextStyle(fontSize: 12.0),
                    )
                  ],
                )
              },
              {
                'color': Colors.white,
                'background': Colors.lightBlue.shade200,
                'label': '3',
                'content': Image.asset(
                  'assets/art.jpg',
                  width: 250,
                  height: 120,
                  fit: BoxFit.cover,
                  alignment: Alignment.center,
                ),
              }
            ],
          ),
        ));
  }
}
```

CHANGELOG.md
## 1.0.1

* Wraps the content with expanded if the direction is set to vertical

## 0.0.1

* Initial release


lib/steps.dart
library steps;

import 'package:flutter/material.dart';

class Steps extends StatelessWidget {
  final Axis direction;
  final List steps;
  final double size;
  final Map path;
  Steps({
    Key key,
    this.direction = Axis.vertical,
    @required this.steps,
    this.size = 21,
    @required this.path,
  }) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return ListView.builder(
      scrollDirection: direction,
      itemCount: steps.length,
      itemBuilder: (context, position) => Stack(
        children: <Widget>[
          Positioned(
            top:
                direction == Axis.horizontal ? 16 + size * 2.14 / 2 - 5 / 2 : 0,
            bottom: direction == Axis.horizontal ? null : 0,
            left:
                direction == Axis.horizontal ? 0 : 16 + size * 2.14 / 2 - 5 / 2,
            right: direction == Axis.horizontal ? 0 : null,
            height: direction == Axis.horizontal ? path['width'] : null,
            width: direction == Axis.horizontal ? null : path['width'],
            child: Container(
              color: path['color'],
            ),
          ),
          Container(
            padding: EdgeInsets.all(16),
            child: Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisSize: MainAxisSize.min,
              children: <Widget>[
                Container(
                  margin: EdgeInsets.only(right: 16),
                  child: Text(
                    steps[position]['label'],
                    style: TextStyle(
                      color: steps[position]['color'],
                      fontSize: size,
                      fontWeight: FontWeight.w900,
                    ),
                  ),
                  padding: EdgeInsets.all(6),
                  width: size * 2.14,
                  height: size * 2.14,
                  alignment: Alignment.center,
                  decoration: BoxDecoration(
                    color: steps[position]['background'],
                    borderRadius: BorderRadius.circular(size * 2.14),
                  ),
                ),
                steps[position]['content'] != null
                    ? direction == Axis.horizontal
                        ? Padding(
                            padding: EdgeInsets.only(top: size * 2.14),
                            child: steps[position]['content'],
                          )
                        : Expanded(
                            child: Padding(
                              padding: EdgeInsets.only(top: size * 2.14),
                              child: steps[position]['content'],
                            ),
                          )
                    : Container(),
              ],
            ),
          )
        ],
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:steps/steps.dart';

void main() => runApp(MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Home(title: 'Steps'),
    ));

class Home extends StatelessWidget {
  final String title;
  Home({Key key, this.title}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text(title),
        ),
        body: Container(
          alignment: Alignment.topCenter,
          child: Steps(
            direction: Axis.vertical,
            size: 20.0,
            path: {'color': Colors.lightBlue.shade200, 'width': 3.0},
            steps: [
              {
                'color': Colors.white,
                'background': Colors.lightBlue.shade200,
                'label': '1',
                'content': Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: <Widget>[
                    Text(
                      'Laborum exercitation',
                      style: TextStyle(fontSize: 22.0),
                    ),
                    Text(
                      'Qui et consectetur esse duis excepteur magna consectetur.',
                      style: TextStyle(fontSize: 12.0),
                    ),
                  ],
                ),
              },
              {
                'color': Colors.white,
                'background': Colors.lightBlue.shade700,
                'label': '2',
                'content': Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: <Widget>[
                    Text(
                      'Laborum exercitation est veniam',
                      style: TextStyle(fontSize: 22.0),
                    ),
                    Image.asset(
                      'assets/cat.jpg',
                      width: 250,
                      height: 120,
                      fit: BoxFit.cover,
                      alignment: Alignment.center,
                    ),
                    Text(
                      '''
Occaecat qui do mollit
Adipisicing reprehenderit deserunt mollit
Quis officia adipisicing aute
                      ''',
                      style: TextStyle(fontSize: 12.0),
                    )
                  ],
                )
              },
              {
                'color': Colors.white,
                'background': Colors.lightBlue.shade200,
                'label': '3',
                'content': Image.asset(
                  'assets/art.jpg',
                  width: 250,
                  height: 120,
                  fit: BoxFit.cover,
                  alignment: Alignment.center,
                ),
              }
            ],
          ),
        ));
  }
}
